File: src/main/java/com/thesilentnights/easylogin/service/TimerService.java
package com.thesilentnights.easylogin.service;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class TimerService {

    private static final Map<String, Long> entries = new HashMap<>();

    public static boolean contains(String identifier) {
        Long expireTime = entries.get(identifier);
        return expireTime != null && System.currentTimeMillis() < expireTime;
    }

    public static String generateIdentifier(UUID uuid, Class<?> serviceClass) {
        return uuid.toString() + "_" + serviceClass.getSimpleName();
    }

    public static void add(String identifier, long duration) {
        entries.put(identifier, System.currentTimeMillis() + duration);
    }

    public void cleanExpired() {
        entries.entrySet().removeIf(entry -> System.currentTimeMillis() > entry.getValue());
    }

    public void clear() {
        entries.clear();
    }
}

File: src/main/kotlin/com/thesilentnights/easylogin/service/TimerService.kt
package com.thesilentnights.easylogin.service

import java.util.*

object TimerService {
    private val entries: MutableMap<String?, Long?> = HashMap<String?, Long?>()

    fun contains(identifier: String?): Boolean {
        return entries[identifier] != null && System.currentTimeMillis() < entries[identifier]!!
    }

    fun cleanExpired() {
        entries.entries.removeIf { entry: MutableMap.MutableEntry<String?, Long?>? -> System.currentTimeMillis() > entry!!.value!! }
    }

    fun generateIdentifier(uuid: UUID?, serviceClass: Class<*>): String {
        return uuid.toString() + "_" + serviceClass.getName()
    }

    fun add(identifier: String?, duration: Long) {
        entries[identifier] = System.currentTimeMillis() + duration
    }

    fun clear() {
        entries.clear()
    }
}
